proxy.go

```diff
@@ -226,12 +268,24 @@ func main() {
 		logger().Fatalf("failed to dial channel(%q): %s", muxAddr, err)
 		return
 	}
-	defer servConn.Close()
+	defer func() {
+		if servConn != nil {
+			servConn.Close()
+			servConn = nil
+		}
+	}()
 
 	results := make(chan error)
-	err = serve(servConn, "unix", listenAddr, results)
+	l, err := serve(servConn, "unix", listenAddr, results)
 	if err != nil {
 		logger().Fatal(err)
+		return
+	}
+
+	// Blocking function waiting for a SIGTERM signal.
+	if err := handleSigtermSignal(sigCh, servConn, l); err != nil {
+		logger().Fatal(err)
+		return
 	}
```

Comment:
It also blocks the following loop of `logger().Fatal(err)`, which should not be blocked.

Reply:
Oh yes, I forgot to start this error printing loop in a go routine, let me change this.

Reply:
Might be best as `logger().WithError(err).Fatal()` to make it nice and parseable too ;)

Reply:
I agree @jodh-intel , but we should do that from an other PR since this should be fixed everywhere in `main()`.

Reply:
yep - wfm